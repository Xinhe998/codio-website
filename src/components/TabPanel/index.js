import PropTypes from 'prop-types';
import React, { Component } from 'react';
import cx from 'classnames';
import './index.scss';

const DEFAULT_CLASS = 'react-tabs__tab-panel';

class TabPanel extends Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    const {
      children,
      className,
      forceRender,
      id,
      selected,
      selectedClassName,
      tabId,
      ...attributes
    } = this.props;
    return (
      <div
        {...attributes}
        className={cx(className, {
          [selectedClassName]: selected,
        })}
        role="tabpanel"
        id={id}
        aria-labelledby={tabId}
      >
        {forceRender || selected ? children : null}
      </div>
    );
  }
}

TabPanel.propTypes = {
  children: PropTypes.node,
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.object,
  ]),
  forceRender: PropTypes.bool,
  id: PropTypes.string, // private
  selected: PropTypes.bool, // private
  selectedClassName: PropTypes.string,
  tabId: PropTypes.string, // private
};
TabPanel.defaultProps = {
  className: DEFAULT_CLASS,
  forceRender: false,
  selectedClassName: `${DEFAULT_CLASS}--selected`,
};
export default TabPanel;
