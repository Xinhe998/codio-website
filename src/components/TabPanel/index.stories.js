import React, { useState } from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { DocsPage, DocsContainer } from '@storybook/addon-docs/blocks';
import StoryRouter from 'storybook-react-router';
import TabPanel from './index';
import storeFs from '../../reducers';

const store = createStore(storeFs);
const stories = storiesOf('TabPanel', module).addParameters({
  title: 'TabPanel',
  component: TabPanel,
  docs: {
    container: DocsContainer,
    page: DocsPage,
  },
});

stories.addDecorator(StoryRouter()).add('default', () => (
  <Provider store={store}>
    <TabPanel selected>Tab 1 Content</TabPanel>
    <TabPanel selected={false}>Tab 2 Content</TabPanel>
  </Provider>
));
