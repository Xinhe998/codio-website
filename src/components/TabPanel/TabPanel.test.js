/* eslint-disable func-names */
import 'babel-polyfill';
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow, configure, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TabPanel from './index';

import fakeImg from '../../assets/smartmockups_jxfuqv8i.jpg';
import fakeAvatar from '../../assets/userImg.png';

import storeFs from '../../reducers';

const store = createStore(storeFs);

// 以該解析器提供給 Enzyme 做渲染 Component 的設置
configure({ adapter: new Adapter() });

describe('TabPanel', function() {
  let testComp;

  beforeEach(() => {
    jest.resetModules();
    this.params = {
      children: 'Test',
      className: 'TestTabPanel',
      selected: true,
      selectedClassName: 'selected',
      tabId: '1',
    };

    this.makeSubject = () => {
      const {
        children,
        className,
        selected,
        selectedClassName,
        tabId,
      } = this.params;

      testComp = mount(
        <Router>
          <Provider store={store}>
            <TabPanel
              className={className}
              selected={selected}
              selectedClassName={selectedClassName}
              tabId={tabId}
            >
              {children}
            </TabPanel>
          </Provider>
        </Router>,
      );
      return testComp;
    };
  });
  describe('when it has complete data', () => {
    beforeEach(() => {
      this.subject = this.makeSubject();
    });
    it('should be same as snapshot', async () => {
      const wrapper = render(await testComp);
      expect(wrapper).toMatchSnapshot();
    });
    it('should find className', async () => {
      expect(testComp.find('.TestTabPanel'));
    });
  });
});
