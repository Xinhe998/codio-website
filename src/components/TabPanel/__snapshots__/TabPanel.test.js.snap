// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanel when it has complete data should be same as snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "aria-labelledby": "1",
      "class": "TestTabPanel selected",
      "role": "tabpanel",
    },
    "children": Array [
      Object {
        "data": "Test",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-labelledby": undefined,
      "class": undefined,
      "role": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-labelledby": undefined,
      "class": undefined,
      "role": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
